packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout/useArrangeAll.ts

```diff
@@ -50,6 +50,16 @@ const elkOptions = {
   'elk.layered.spacing.edgeNodeBetweenLayers': '40',
 };
 
+const computeHeaderVerticalFootprint = (node, viewportZoom: number): number => {
+  if (node && node.data.insideLabel?.isHeader) {
+    const label = document.querySelector<HTMLDivElement>(`[data-id="${node.data.insideLabel.id}-content"]`);
```

Comment:
Wouldn't that create errors with diagram in a portal? Something like `reactFlowDomElement.querySelector(...)` would be better to not start from the root of the web page. I think we had this issue somewhere else in the code of Reactflow (in the palette IIRC).

Reply:
I changed it to use the ref of the `DiagramRender`